index.js

```diff
@@ -1 +1,214 @@
+     */
+    constructor(emitter, type) {
+        this.emitter = emitter;
+        this.type = type;
+        this.events = new EventEmitter();
+        this.handler = (...args) => {
+            const errors = [];
+
+            // Iterating a copy of the handlers guarantees any mutation of the handler collection
+            // will only take affect the next time this event is emitted
+            [ ...this.handlers ].forEach(
+                (handler) => {
+                    // Run each handler in it's own quarantine. Collect errors
+                    try {
+                        handler.apply(this.emitter, args);
+                    } catch (error) {
+                        errors.push(error);
+                    }
+                }
+            );
+            if (errors.length) {
+
+                // avoid unhandledRejection loop if no handler was registered
+                if (
+                    this.emitter === process &&
+                    /unhandledRejection/i.test(this.type) &&
+                    this.events.listenerCount('error') === 0
+                ) {
+                    this.events.on('error', (error) => console.error(error));
+                }
+
+                // Running over all handlers for each error O(n²)
+                errors.forEach(
+                    (error) => this.events.emit('error', error)
+                );
+            }
+        };
+    }
+
+    /**
+     * Reinstate native event listener
+     * @returns {self}
+     */
+    unmount() {
+        if (!this.mounted) {
+            return this;
+        }
+
+        Object.entries(this.mounted).forEach(
+            ([ key, value ]) => {
+                this.emitter[key] = value;
+                delete this.mounted[key];
+            }
+        );
+        delete this.mounted;
+
+        this.emitter.removeListener(
+            this.type,
+            this.handler
+        );
+
+        // Re attach to emitter as individual event handlers
+        this.handlers.forEach(
+            (handler) => this.emitter.on(this.type, handler)
+        );
+
+        return this;
+    }
+
+    /**
+     * Override organic event listeners
+     * @returns {self}
+     */
+    mount() {
+        if (this.mounted) {
+            return this;
+        }
+        this.handlers = this.emitter.listeners(this.type);
+        this.mounted = {};
+
+        // Remove all existing listeners
+        this.emitter.removeAllListeners(this.type);
+
+        // Engulf existing listener which controls the handlers execution
+        this.emitter.on(
+            this.type,
+            this.handler
+        );
+
+        this.override({
+            aliases: [ 'addListener', 'on' ],
+            implementation: (type, handler, { once } = {}) => {
+                once
+                    ? this.emitter.once(type, handler)
+                    : this.handlers.push(handler)
+                ;
+            }
+        });
```

Comment:
Not sure I fully understand the purpose of overriding, mind elaborating a little bit?

Reply:
Added a paragraph in the readme
> ## Reason
> By overriding native behaviour we can verify existing event handlers run in a safe environment, within a try/catch block. This way we can avoid unexpected results, such as the process exiting unexpectedly within an event handler. We can later decide how we want to handle these errors by placing a general onerror handler on the custodian.

Do you think it clarifies the purpose?